frontend/src/js/external-forms/stateSelectors.ts

```diff
@@ -31,22 +32,23 @@ export const selectRunningQuery = (state: StateT) => {
 };
 
 function getVisibleConceptListFields(
-  config: { fields: GeneralField[] },
+  config: Form,
+  fields: GeneralField[],
```

Comment:
So if I understand your code right, this `fields` is global, for ALL available forms at once? If that's the case, why don't you try reworking fields to be a map `config["type"] -> GeneralField[]` (since that's what you accomplished in effect either way). Thereby avoiding having to virtualise the field-name.